src/main/java/com/cinema/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.cinema.service.impl;
+
+import com.cinema.dao.OrderDao;
+import com.cinema.dao.ShoppingCartDao;
+import com.cinema.lib.Inject;
+import com.cinema.lib.Service;
+import com.cinema.model.Order;
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.Ticket;
+import com.cinema.model.User;
+import com.cinema.service.OrderService;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        List<Ticket> tickets = shoppingCart.getTickets();
+        User user = shoppingCart.getUser();
+        Order order = new Order();
+        order.setUser(user);
+        order.setTickets(tickets);
+        order.setOrderDate(LocalDateTime.now());
+        orderDao.add(order);
+        shoppingCart.setTickets(new ArrayList<>());
+        shoppingCartDao.update(shoppingCart);
```

Comment:
It's okay to user one service inside other. Let's use ShoppingCartService.clear() here/

Reply:
done